theworkoutinfo/ViewModels/HomeViewModel.swift

```diff
@@ -9,32 +9,43 @@
 import Foundation
 import RxSwift
 
+public typealias Animate = (wantAnimate: Bool, message: String?)
+
 class HomeViewModel {
     
     let disposeBag = DisposeBag()
     
     //MARK: - Inputs
     
     let tryLogout: AnyObserver<Void>
-    let loadExercises: AnyObserver<Int>
+    let loadExercises: AnyObserver<(Bool,Int)>
     let selectExercise: AnyObserver<Exercise>
+    let loadExercisesByCategory: AnyObserver<(Bool,ExerciseCategory,Int)>
+    let loadCategories: AnyObserver<Void>
+    let tryAnimate: AnyObserver<Animate>
     
     //MARK: - Outputs
     
+    let isAnimating: Observable<Animate>
     let didLogout: Observable<Void>
     let didSelectExercise: Observable<Exercise>
     var exercises: Observable<[Exercise]> {
         return _exercisesVariable.asObservable()
     }
     
+    var categories: Observable<[ExerciseCategory]> {
+        return _categoriesVariable.asObservable()
+    }
+    
     var error: Observable<String> {
         return _error.asObservable()
     }
     
     //MARK: - Properties
-        
+    
     private let _error = PublishSubject<String>()
     private let _exercisesVariable = Variable<[Exercise]>([])
+    private let _categoriesVariable = Variable<[ExerciseCategory]>([ExerciseCategory(id: -1, name: "All")])
```

Comment:
Does categories need to be a variable? Why not an Observable? If you need to use an initial value for the category -1 consider using a BehaviorSubject instead of a PublicSubject

Reply:
Also, this array with just the "-1" category appears twice, so I'd put it in another place

Reply:
Maybe a variable is need, let's discuss this later